src/components/inputs/CPriceInput.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <v-text-field
+    v-model="fieldValue"
+    :append-inner-icon="mdiCurrencyEur"
+    :label="label"
+    :variant="variant"
+    :disabled="disabled"
+    :rules="rules"
+    placeholder="0,00"
+    color="primary"
+    @input="onInputChange"
+  ></v-text-field>
+</template>
+
+<script setup lang="ts">
+import { mdiCurrencyEur } from "@mdi/js";
+import { ref, watch } from "vue";
+
+/**
+ * The currently entered price.
+ */
+const model = defineModel<number | undefined>({ required: true });
+
+/**
+ * The model for the input field.
+ *
+ * This field is just a ref and will only be updated from the field itself and the watcher of the `model`.
+ *
+ * A computed ref is not used because the parse functions will automatically remove any disallowed chars on change.
+ * This causes the `model`s value to not change and therefore, the computed not to be updated, which leads to the problem
+ * that disallowed chars are not removed anymore.
+ */
+const fieldValue = ref<string | undefined>();
+
+/**
+ * Watches the model to set the newest value to `fieldValue`.
+ *
+ * This one will only use `formatPrice` to also support database formats like `12` to be parsed to `12,00` instead of `0,12`.
+ */
+watch(
+  model,
+  () => {
+    // Update field value to fit format
+    fieldValue.value = model.value
+      ? model.value.toFixed(2).toString().replace(".", ",")
```

Comment:
Please decide the decimal separator based on the locale

Reply:
I think we should add localization later since we would also need to make the currency symbol dynamic then wouldn't we?

Or is this not dependent on the currency but only on the language?

Reply:
They are related but multi currency support is not a goal yet. Multi language support is a requirement though.